modules/library/main/src/main/java/org/geotools/styling/Histogram.java

```diff
@@ -0,0 +1,45 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2015, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.styling;
+
+import org.opengis.filter.FilterFactory;
+import org.opengis.style.ContrastMethod;
+
+/**
+ * @author iant
+ *
+ */
+public class Histogram extends AbstractContrastEnhancementMethod {
```

Comment:
These are looking more and more like a strategy (the API polite way of saying algorithm).

So this would become:

````
public class HistogramContrastEnhancementStratagy extends AbstractContrastEnhancementStratagy {
````

Failing that we have our naming convention for implementations making this HistogramContrastEnhancementMethod because java coding is really that verbose.